src/librustc/ty/structural_impls.rs

```diff
@@ -134,7 +134,7 @@ impl<'a, 'tcx> Lift<'tcx> for ty::ProjectionTy<'a> {
     type Lifted = ty::ProjectionTy<'tcx>;
     fn lift_to_tcx<'b, 'gcx>(&self, tcx: TyCtxt<'b, 'gcx, 'tcx>)
                              -> Option<ty::ProjectionTy<'tcx>> {
-        tcx.lift(&self.trait_ref).map(|trait_ref| {
+        tcx.lift(&self.trait_ref(tcx)).map(|trait_ref| {
             ty::ProjectionTy::from_ref_and_name(tcx, trait_ref, self.item_name(tcx))
```

Comment:
This should lift substitutions instead. `from_ref_and_name` should likely not exist.